src/common/schedulers/slurm_commands.py

```diff
@@ -171,12 +174,40 @@ def _recompute_required_nodes_by_gpu_reservation(pending_jobs, gpus_per_node):
     for job in pending_jobs:
         gpus_per_job = job.tres_per_job.get("gpu")
         if gpus_per_job:
-            job.nodes = max(int(math.ceil(float(gpus_per_job) / gpus_per_node)), job.nodes)
+            new_min_nodes = int(math.ceil(float(gpus_per_job) / gpus_per_node))
+            _update_job_nodes_and_cpus_total(job, new_min_nodes)
 
         gpus_per_task = job.tres_per_task.get("gpu")
         if gpus_per_task:
             tasks_schedulable_per_node = float(gpus_per_node) // gpus_per_task
-            job.nodes = max(int(math.ceil(float(job.tasks) / tasks_schedulable_per_node)), job.nodes)
+            new_min_nodes = int(math.ceil(float(job.tasks) / tasks_schedulable_per_node))
+            _update_job_nodes_and_cpus_total(job, new_min_nodes)
+
+
+def _validate_cpus_total_with_gpus_total(job, gpus_total):
```

Comment:
s/validate/recompute

Reply:
renamed to `_recompute_cpus_total_with_gpus_total`